File: android/src/main/java/com/jasonasmk/clarityx/ClarityPluginXPlugin.java
package com.jasonasmk.clarityx;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "ClarityPluginX")
public class ClarityPluginXPlugin extends Plugin {


    @PluginMethod
    public void initialize(PluginCall call) {
        String projectId = call.getString("projectId");
        ClarityPluginX implementation = new ClarityPluginX(getActivity());

        Boolean result = implementation.initialize(projectId);

        if(result) {
            call.resolve();
            return;
        }
        call.reject("Clarity initialization failed");
    }

    @PluginMethod
    public void setCustomUserId(PluginCall call) {
        String customUserId = call.getString("customUserId");
        ClarityPluginX implementation = new ClarityPluginX(getActivity());

        Boolean result = implementation.setCustomUserId(customUserId);

        if(result) {
            call.resolve();
            return;
        }
        call.reject("Clarity setCustomUserId failed");
    }
}

File: ios/Sources/ClarityPluginXPlugin/ClarityPluginXPlugin.swift
import Foundation
import Capacitor
/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(ClarityPluginXPlugin)
public class ClarityPluginXPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "ClarityPluginXPlugin"
    public let jsName = "ClarityPluginX"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "initialize", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = ClarityPluginX()

    @objc func initialize(_ call: CAPPluginCall) {
        let projectId = call.getString("projectId") ?? ""
        let result = implementation.initialize(projectId)
        
        if(result) {
            call.resolve()
        } else {
            call.reject("Error initializing Clarity")
        }
    }
    
    @objc func setCustomUserId(_ call: CAPPluginCall) {
        let customUserId = call.getString("customUserId") ?? ""
        let result = implementation.setCustomUserId(customUserId)
        if(result) {
            call.resolve()
        } else {
            call.reject("Error setting custom user id")
        }
    }
}
